### PR TITLE
Add headless Amplify cli support

### PR DESCRIPTION
**Pull request by salihgueler**

### CODE REVIEW

**Review by felangel — COMMENTED** [review]
Thoughts on removing the .DS_Store? I think it can be gitignored

File: hooks/.dart_tool/package_config.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
+{
```
Comment (felangel) [inline hooks/.dart_tool/package_config.json]: The .dart_tool directory should be gitignored as well

File: hooks/.packages [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# This file is deprecated. Tools should instead consume
```
Comment (felangel) [inline hooks/.packages]: This file should be gitignored as well

File: hooks/post_gen.dart [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  final input = stdin.readLineSync() ?? '';
+
+  _runAmplifyPush(context, input);
+}
+
+void _runFlutterPubGet(HookContext context) {
+  final flutterPubGetProgress = context.logger.progress(
+    'running "flutter pub get"',
+  );
+  final result = Process.runSync(
```
Comment (felangel) [inline hooks/post_gen.dart]: Thoughts on using Process.start for consistency?

File: hooks/post_gen.dart [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    exit(result.exitCode);
+  }
+}
+
+void _runAmplifyInit(HookContext context) {
+  final amplifyInitProgress = context.logger.progress(
+    'running "amplify init --yes"',
+  );
+
+  final amplifyResult = Process.runSync(
```
Comment (felangel) [inline hooks/post_gen.dart]: Same here

File: hooks/pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+name: amplify_starter_hooks
+
+environment:
+  sdk: ">=2.12.0 <3.0.0"
+
+dependencies:
+  mason: ^0.1.0-dev.28
```
Comment (felangel) [inline hooks/pubspec.yaml]: This can be bumped to the latest version as well

File: hooks/post_gen.dart (hunk 3 above)
Comment (salihgueler) [reply hooks/post_gen.dart]: Makes sense 👍🏼

**Comment by salihgueler** [comment]
@felangel All comments are addressed!

**Comment by felangel** [comment]
> @felangel All comments are addressed!

Looks like the `.DS_Store` is still there but otherwise LGTM!

File: .gitignore [hunk 6]
```diff
@@ -1 +1,3 @@
-.idea/
\ No newline at end of file
+.idea/
+hooks/.dart_tool
```
Comment (felangel) [inline .gitignore]:
```suggestion
```

File: .gitignore [hunk 7]
```diff
@@ -1 +1,3 @@
...
+hooks/.packages
```
Comment (felangel) [inline .gitignore]:
```suggestion
```

Comment (felangel) [reply .gitignore]: I don’t think those are necessary since you added them to the hooks .gitignore

Comment (salihgueler) [reply .gitignore:3]: > .DS_Store

That is odd, I don't see it in my system. But I added to the project's gitignore now.

**Comment by salihgueler** [comment]
> .DS_Store

That is odd, I don't see it in my system. But I added to the project's gitignore now. EDIT: When I added the file to git ignore it showed itself in the VSCode but still not in the file system hahaha

**Comment by salihgueler** [comment]
Screenshot 2022-08-04 at 17 03 26

**Comment by felangel** [comment]
Weird...it's still showing up in the diff.
Did you try `git rm --cached .DS_Store`?

Screen Shot 2022-08-04 at 9 36 37 AM

**Comment by salihgueler** [comment]
> Weird...it's still showing up in the diff. Did you try `git rm --cached .DS_Store`?
> 
> Screen Shot 2022-08-04 at 9 36 37 AM

How about now? 😄

**Review by felangel — APPROVED** [review]

**Comment by felangel** [comment]
> > Weird...it's still showing up in the diff. Did you try `git rm --cached .DS_Store`?
> > Screen Shot 2022-08-04 at 9 36 37 AM
> 
> How about now? 😄

All good now 🎉

**Pull request merged**